onflow/fcl-js
# Nialexsan/1542 switch to cross fetch

#1542

switch to cross-fetch from node-fetch

cross-fetch works as ponyfil (isomorphic fetch doesn't):
https://www.npmjs.com/package/cross-fetch

## Testing

1. install packages and build the repo
> npm i && npm run build

2. pack fcl and transport http
> cd ./packages/fcl && npm pack && ../transport-http && npm pack

3. copy generated archives to `fcl-nextjs-quickstart` project
https://github.com/muttoni/fcl-nextjs-quickstart

4. change dependencies location:
_(version suffix could be different)_
```
-    "@onflow/fcl": "^1.3.2",
+    "@onflow/fcl": "file:onflow-fcl-1.4.0-alpha.10.tgz",
+    "@onflow/transport-http": "file:onflow-transport-http-1.6.0-alpha.5.tgz",
```

5. remove `resolve` block from `vite.config.js`
```
-    resolve:{
-        alias:{
-            "node-fetch": "./node_modules/node-fetch/browser.js"
-        }
-    },
```
6. run the project
> rm -rf node_modules && rm package-lock.json && npm i && npm run dev -- --open
7. verify that fcl still works and there are no fcl related errors in the console
